### PR TITLE
Add `--disable-admin-operations` flag in Compactor UI and Bucket UI

### LINKED ISSUE #5390: Provide option to disable block mark deletion and mark no compaction from bucket UI
In Prometheus, admin level functionalities like delete series are not provided through UI.
But in Thanos, now it is supported to mark deletion and mark no compaction from bucket UI. This is an admin-level operation and requires admin permissions. Some users might not want to expose those functionalities from bucket UI to their users.

We can have an option to disable those functionalities from UI or make those features not enabled by default.

### PR DESCRIPTION
**Pull request by harsh-ps-2003**
Signed-off-by: Harsh Pratap Singh <harshpratapsingh8210@gmail.com>
Fixes #5390 

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This PR adds a flag, named `--disable-admin-operations`, to the Compactor UI and Bucket Web UI to decide whether to disable the corresponding buttons Mark Deletion and Mark No Compaction in the UI as they are admin-level operations. It also restricts the API usage for both of them.

## Verification

It was tested locally.
The UI test:
Screenshot 2023-08-10 at 9 09 08 PM
Screenshot 2023-08-10 at 9 09 20 PM
Screenshot 2023-08-10 at 9 08 42 PM
Screenshot 2023-08-10 at 9 08 52 PM

The API test:
```
❯ curl -X POST -d "id=_ulid&action=DELETION" http://localhost:8080/api/v1/blocks/mark
{"status":"error","errorType":"bad_data","error":"Admin operations are disabled"}
❯ curl -X POST -d "id=_ulid&action=DELETION" http://localhost:10922/api/v1/blocks/mark
{"status":"error","errorType":"bad_data","error":"Admin operations are disabled"}
```
cc @yeya24 @GiedriusS @saswatamcode

### CODE REVIEW

**Comment by harsh-ps-2003** [comment]
If everything goes okay with this PR, I will add the changelog. Sorry for messing up a lot of times and thanks for the patience. Also how to fix this Docs CI failing!?

**Comment by saswatamcode** [comment]
@harsh-ps-2003 you don't need to raise a new PR every time after review. you can push commits to address feedback. 🙂 
Also, you can run `make doc` and commit to fix docs CI

File: cmd/thanos/compact.go [hunk 1]
```diff
@@ -786,4 +787,6 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 	cc.webConf.registerFlag(cmd)
 
 	cmd.Flag("bucket-web-label", "External block label to use as group title in the bucket web UI").StringVar(&cc.label)
+
+	cmd.Flag("disable-admin-operations", "Restrict access to Block Mark deletion and no compaction").Default("false").BoolVar(&cc.disableAdminOperations)
```
Comment (saswatamcode) [inline cmd/thanos/compact.go:791]: Let's make description a bit clearer.
```suggestion
	cmd.Flag("disable-admin-operations", "Disable UI/API admin operations like marking blocks for deletion and no compaction.").Default("false").BoolVar(&cc.disableAdminOperations)
```

File: cmd/thanos/tools_bucket.go [hunk 2]
```diff
@@ -203,6 +204,8 @@ func (tbc *bucketWebConfig) registerBucketWebFlag(cmd extkingpin.FlagClause) *bu
 	cmd.Flag("timeout", "Timeout to download metadata from remote storage").Default("5m").DurationVar(&tbc.timeout)
 
 	cmd.Flag("label", "External block label to use as group title").StringVar(&tbc.label)
+
+	cmd.Flag("disable-admin-operations", "Restrict access to Block Mark deletion and no compaction").Default("false").BoolVar(&tbc.disableAdminOperations)
```
Comment (saswatamcode) [inline cmd/thanos/tools_bucket.go:208]: ditto
```suggestion
	cmd.Flag("disable-admin-operations", "Disable UI/API admin operations like marking blocks for deletion and no compaction.").Default("false").BoolVar(&tbc.disableAdminOperations)
```

File: pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx [hunk 3]
```diff
@@ -100,26 +105,30 @@ export const BlockDetails: FC<BlockDetailsProps> = ({ block, selectBlock }) => {
...
+                onClick={() => {
+                  setModalAction('DELETION');
+                  setDetailValue('');
+                }}
+              >
+                Mark Deletion
+              </Button>
+            </div>
+          )}
+          {!disableAdminOperations && (
```
Comment (saswatamcode) [inline pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx:120]: We can probably render both buttons, under a single conditional?

File: pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx [hunk 4]
```diff
@@ -40,6 +42,9 @@ export const BlockDetails: FC<BlockDetailsProps> = ({ block, selectBlock }) => {
     }
   };
 
+  const { response: flagsRes } = useFetch<FlagMap>(`/api/v1/status/flags`);
+  const disableAdminOperations = flagsRes?.data?.['disable-admin-operations'] || false;
```
Comment (saswatamcode) [inline pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx:46]: Let's actually call this in BlocksContent and pass it to BlocksDetails as a prop, that way we won't be calling this every time we open a block detail page. This also won't change from block to block.

**Review by saswatamcode — APPROVED** [review]
Thanks! 
Generally LGTM mod some small nits. 🙂 

Let's try to address comments here instead of opening new PR!

**Comment by harsh-ps-2003** [comment]
Thanks for the review and sorry for the closed PRs @saswatamcode

**Comment by harsh-ps-2003** [comment]
@saswatamcode @yeya24 @GiedriusS when I did `make docs`, it gave me an error 
```
>> cleaning up white noise
sed: 1: "./CODE_OF_CONDUCT.md": invalid command code .
make[1]: *** [white-noise-cleanup] Error 1
make: *** [docs] Error 2
```
as I am on Mac OS. The invalid `sed` command is happening as on macOS; the `sed` command requires an empty string argument after the `-i` option when doing in-place editing, while on Linux, this argument is optional. I have locally made changes to the `cleanup-white-noise` script to work on both Mac and Linux systems. And this was the reason the Docs CI was failing. 
After fixing this when I do `make docs` I don't get any errors, but when I do `make check-docs` I get some errors! Working on it...

**Pull request closed without merging**